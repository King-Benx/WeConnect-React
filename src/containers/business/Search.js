import React from 'react';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types'

class Search extends React.Component{
    constructor(props){
        super(props);
        this.state = {search:"", filter_type:"",filter_value:"",description:""}
    }

    static contextTypes = {
        router: PropTypes.object
    }
    
    handleChange = (event) => {
        this.setState({[event.target.id]: event.target.value});
    }

    formSubmit = (event) => {
        event.preventDefault();
        const search_query= "/search/"+this.state.search+"/"+this.state.filter_type+"/"+this.state.filter_value;
        this.context.router.history.push(search_query)
        this.setState({search:"", filter_type:"",filter_value:"",description:""})
    }

    render(){
        return (
        <li>
            <form className="navbar-form" onSubmit={ this.formSubmit }>
                <div className="form-group">
                    <select id="filter_type" name="filter_type" value={ this.state.filter_type } className="form-control" onChange={ this.handleChange }>
                        <option value="">Filter by</option>
                        <option value="location">Location</option>
                        <option value="category">Category</option>
                    </select>|
                </div>
                <div className="form-group">
                    <input type="text" id="filter_value" value={ this.state.filter_value } onChange={ this.handleChange } className="form-control" placeholder="filter value"/>|
                </div>
                <div className="form-group">
                    <input type="text" id="search" value={ this.state.search } onChange={ this.handleChange } className="form-control" placeholder="Search" required="required"/>|
                </div>
                <Button bsStyle="info" type="submit" onChange={ this.handleChange } className="btn btn-default"><i className="glyphicon glyphicon-search"></i></Button>
            </form>
        </li>
        );
    }
}
export default Search;