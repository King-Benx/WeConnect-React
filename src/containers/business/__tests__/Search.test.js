import React from "react";
import { shallow } from "enzyme";
import Search from "../Search";
import { BrowserRouter } from "react-router-dom";

describe("These are tests for the search form ", () => {
  const event = {
    target: {
      value: {}
    },
    preventDefault: () => {}
  };
  let mountedSearch;
  beforeEach(() => {
    mountedSearch = shallow(<Search />);
  });

  it("renders without crashing", () => {
    mountedSearch;
  });

  it("handles change", () => {
    mountedSearch.instance().handleChange(event);
  });

  it("has an li", () => {
    const lis = mountedSearch.find("li");
    expect(lis.length).toBe(1);
  });

  it("has a form", () => {
    const forms = mountedSearch.find("form");
    expect(forms.length).toBe(1);
  });

  it("has divs", () => {
    const divs = mountedSearch.find("div");
    expect(divs.length).toBe(3);
  });

  it("has a select", () => {
    const selects = mountedSearch.find("select");
    expect(selects.length).toBe(1);
  });

  it("has inputs", () => {
    const inputs = mountedSearch.find("input");
    expect(inputs.length).toBe(2);
  });

  it("has options", () => {
    const options = mountedSearch.find("option");
    expect(options.length).toBe(3);
  });

  it("has a button", () => {
    const buttons = mountedSearch.find("Button");
    expect(buttons.length).toBe(1);
  });

  it("has i_s", () => {
    const i_s = mountedSearch.find("i");
    expect(i_s.length).toBe(1);
  });
});
