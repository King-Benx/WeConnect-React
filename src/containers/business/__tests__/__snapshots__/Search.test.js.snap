// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the search form  renders without crashing 1`] = `
<li>
  <form
    className="navbar-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <select
        className="form-control"
        id="filter_type"
        name="filter_type"
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          Filter by
        </option>
        <option
          value="location"
        >
          Location
        </option>
        <option
          value="category"
        >
          Category
        </option>
      </select>
      |
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="filter_value"
        onChange={[Function]}
        placeholder="filter value"
        type="text"
        value=""
      />
      |
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        id="search"
        onChange={[Function]}
        placeholder="Search"
        required="required"
        type="text"
        value=""
      />
      |
    </div>
    <button
      className="btn btn-default btn btn-info"
      disabled={false}
      type="submit"
    >
      <i
        className="glyphicon glyphicon-search"
        type="submit"
      />
    </button>
  </form>
</li>
`;
